### DISCUSSION TITLE
Filter AAAA but the opposite

### DISCUSSION THREAD

**Discussion by shrunbr** [TechnitiumSoftware/DnsServer#1016] [Ideas]
Hiya!

Currently the "Filter AAAA" app blocks all AAAA records unless you allow a bypass per network/domain. I propose an option to have it do the reverse where it allows all AAAA records except for the domains specified.

This would come in handy for those of us using [Hurricane Electric's TunnelBroker](https://tunnelbroker.net/) to be able to block AAAA records for streaming services such as Netflix, Hulu, etc who have blocked/flagged HE's IPv6 networks due to them being a workaround for VPNs.

For example, doing this would allow me to have the server resolve AAAA records but force the specified domains to return NODATA and then fallback to the A record thus using IPv4 to get to the services.

_I could not find anyone requesting this in either an issue or discussion so I apologize if it's already been posted about._

**Comment by ShreyasZare** [comment]
Thanks for the post. Will evaluate this to see how such an option can be added to filter AAAA for only specified domain names. Will update here once done.

**Reply by IngmarStein** [reply]
This would also help with https://github.com/TechnitiumSoftware/DnsServer/discussions/1070#discussioncomment-10908857: if I could enable Filter AAAA only for `localdomain` and `example.com`, I could avoid the side-effects for anything else.

**Reply by ShreyasZare** [reply]
@IngmarStein yes, I am planning to update the app soon.

**Reply by IngmarStein** [reply]
Excellent. I'd like to give something back for the great support and it looks to me that better documentation could have helped in several support requests. Would you appreciate if I started documenting the `Filter AAAA` app and others? If so, what would be the preferred format? Markdown files in the app directories? GitHub wiki?

**Reply by ShreyasZare** [reply]
That's true, documentation is really needed but unfortunately due to lack of time, I am unable to write it. It would be great if you can contribute to document anything really. 

Earlier the documentation was published on the website but I realized that it would be better to have it in the GitHub repo itself as markdown which would make it easier to write, maintain, and also for anyone to refer/contribute. I would not prefer GitHub Wiki since simple md files would be much more flexible and wont depend on a specific vendor's service.

If you can do start it then that the that would be really appreciated. I would recommend that you start it with a "Docs" folder in the repo. Later, it can be expanded with an Index page which would link to various other markdown pages.

**Comment by ShreyasZare** [comment]
I have released FilterAAAA v3 app which adds new `filterDomains` option in the config thanks to PR by @IngmarStein. You can take a look at the app's [readme file](https://github.com/TechnitiumSoftware/DnsServer/blob/develop/Apps/FilterAaaaApp/README.md) to understand how the new feature works.
